app/javascript/redux/greeting/greeting.js

```diff
@@ -0,0 +1,22 @@
+import fetchGreeting from "../../components/fetchGreeting";
+
+const FETCH_GREETING = 'helloworld/greeting/GET_GREETING'
+
+export const getGreeting = () => async (dispatch) => {
+  const data = await fetchGreeting()
+  dispatch({
+    type: FETCH_GREETING,
+    data
+  })
+}
+
+const greetingReducer = (state = {}, action) => {
+  switch (action.type) {
+    case FETCH_GREETING:
+      return { ...state, ...action.data };
+      default:
+      return state;
+  }
+}
+
+export default greetingReducer;
```

Comment:
- [ ] I tried to locate the component folder you made import from, I could not find it, in fact, the code is yet to be complete here. I would encourage you to follow the steps I took in the code below, not exactly, but your code for this file should be something similar
```
import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import axios from "axios";

const initialState = {
  greeting: '',
};

export const fetchGreetings = createAsyncThunk('greeting/fetchGreetings', async () => {
  try {
    const res = await axios.get('/greetings', {
      headers: {
        'Accept': 'application/json',
      },
    });
    console.log(res)
    return res.data;
  } catch (err) {
    return err.message
  }
})

const greetingSlice = createSlice({
  name: 'greeting',
  initialState,
  reducers: {

  },
  extraReducers: (builder) => {
    builder
    .addCase(fetchGreetings.fulfilled, (state, { payload }) => {
      let randomIndex = Math.floor(payload.length * Math.random());
      state.greeting = payload[randomIndex].text;
    })
    .addCase(fetchGreetings.rejected, (state, { payload }) => {
      state.greeting = 'Fetch data failed';
    })
  }
});

export default greetingSlice.reducer;
```